src/signal_routing.rs

```diff
@@ -52,6 +52,59 @@ pub fn route_action(
     route_signal(action, signal, action.name().as_str(), target)
 }
 
+fn panic_if_signal_cannot_be_queued(signal_name: &str, parameters: &[glib::Value]) {
+    for (i, param) in parameters.iter().enumerate() {
+        let param_type = param.type_();
+        if param_type.name().ends_with("Context") {
+            panic!(
+                concat!(
+                    "Signal {:?}'s param at position {} is a {}. ",
+                    "Signals with context params cannot be invoked from inside the Actix runtime. ",
+                    "Try running whatever triggered it with `woab::outside()` or `woab::spawn_outside()",
+                ),
+                signal_name, i, param_type,
+            );
+        }
+    }
+}
+
+fn run_signal_routing_future(
+    future: impl core::future::Future<Output = Result<Result<Option<gtk::Inhibit>, crate::Error>, actix::MailboxError>> + 'static,
+    signal_name: &Rc::<String>,
+    parameters: &[glib::Value],
+) -> Option<glib::Value> {
+    match crate::try_block_on(future) {
+        Ok(result) => {
+            let result = result.unwrap().unwrap();
+            if let Some(gtk::Inhibit(inhibit)) = result {
```

Comment:
Nit: I'd say something like `result.map(|inhibit| inhibit.0).map(ToValue::to_value)` would be more concise. Or maybe `result.map(|inhibit| inhibit.0.to_value())`. (I think `Inhibit` lacks a good API to get to the wrapped value. But also, this is a rather rare corner case.)